platform/ios/src/MGLMapView.mm

```diff
@@ -1827,6 +1827,10 @@ - (MGLMapCamera *)cameraByZoomingToZoomLevel:(double)zoom  aroundAnchorPoint:(CG
     currentCameraOptions.anchor = anchor;
     camera = [self cameraForCameraOptions:currentCameraOptions];
     
+    CLLocationCoordinate2D centerCoordinate = [self convertPoint:anchorPoint toCoordinateFromView:view];
+    
+    camera.centerCoordinate = centerCoordinate;
```

Comment:
This change conflates the anchor point of a zoom with the map view’s center coordinate: https://github.com/mapbox/mapbox-gl-native/issues/10391#issuecomment-343388563.